sycl/source/detail/device_filter.cpp

```diff
@@ -17,60 +17,76 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace detail {
 
+std::vector<std::string> tokenize(const std::string &Filter,
+                                  const std::string &Delim) {
+  std::vector<std::string> Tokens;
+  size_t Pos = 0;
+  std::string Input = Filter;
+  std::string Tok;
+
+  while ((Pos = Input.find(Delim)) != std::string::npos) {
+    Tok = Input.substr(0, Pos);
+    Input.erase(0, Pos + Delim.length());
```

Comment:
It would be more efficient to not modify the input string. I suggest we use some the code similar to existing implementation or `std::string_view`.

I would also change the return type to `std::vector<std::string_view>`.

Reply:
`Input` is a local copy of the parameter 'Filter', not directly changing the parameter 'Filter'.

Reply:
My point is about efficiency. Making local copy of `std::string` input parameter and modifying this string is not necessary for the task this function is doing.

Reply:
Please note that this function `tokenize` is modifying the string, which is not suitable for std::string_view.
This function is extracting and erasing substring.
Further returning a string_view from a locally created substring is not going to work because the local substring will be out of scope after returning from this function, and string_view to it is invalid.

Reply:
That's why we should not create local substrings. Returned `string_view`s should reference `Filter` string.

Reply:
I see. Instead of creating a local substring copy, we can create a substring_view. I will do that. Thanks.

Reply:
Done

Reply:
> I see. Instead of creating a local substring copy, we can create a substring_view. I will do that. Thanks.
> Done.

Sorry, I don't see these changes. Did you forget to push them to the remote?
I also would like to note that this refactoring can be applied in a separate PR as this is not directly related to making device ids unique per backend.

Reply:
Sorry, somehow I forgot about this request. :)

Reply:
Done.